api/lib/source/polkadotV0-source.js

```diff
@@ -529,6 +534,166 @@ class polkadotAPI {
       }
     )
   }
+
+  async getAllProposals() {
+    const api = await this.getAPI()
+
+    const [
+      blockHeight,
+      totalIssuance,
+      democracyProposals,
+      democracyReferendums,
+      treasuryProposals,
+      councilProposals,
+      councilMembers
+    ] = await Promise.all([
+      this.getBlockHeight(),
+      api.query.balances.totalIssuance(),
+      api.derive.democracy.proposals(),
+      api.derive.democracy.referendums(),
+      api.derive.treasury.proposals(),
+      api.derive.council.proposals(),
+      api.query.council.members()
+    ])
+    const allProposals = democracyProposals
+      .map((proposal) => {
+        return this.reducers.democracyProposalReducer(
+          this.network,
+          proposal,
+          totalIssuance,
+          blockHeight
+        )
+      })
+      .concat(
+        democracyReferendums.map((proposal) => {
+          return this.reducers.democracyReferendumReducer(
+            this.network,
+            proposal,
+            totalIssuance,
+            blockHeight
+          )
+        })
+      )
+      .concat(
+        treasuryProposals.proposals.map((proposal) => {
+          return this.reducers.treasuryProposalReducer(
+            this.network,
+            proposal,
+            councilMembers
+          )
+        })
+      )
+      .concat(
+        councilProposals.map((proposal) => {
+          return this.reducers.councilProposalReducer(
+            this.network,
+            proposal,
+            councilMembers,
+            blockHeight
+          )
+        })
+      )
+
+    return orderBy(allProposals, 'id', 'desc')
+  }
+
+  getProposalById() {}
+
+  getDelegatorVote() {}
+
+  async getTotalActiveAccounts() {
+    const api = await this.getAPI()
+    const accountKeys = await api.query.system.account.keys()
+    const accounts = accountKeys.map((key) => key.args[0].toHuman())
+    return accounts.length || 0
+  }
+
+  async getRecentProposals() {
+    // while we don't have proposals in DB this is the only way
+    const proposals = await this.getAllProposals()
+    // the problem in Substrate is that right now we don't have all creationTimes yet
+    return proposals.sort(
+      (a, b) => new Date(b.creationTime) - new Date(a.creationTime)
+    )
+  }
+
+  async pollForTopVoters() {
```

Comment:
why do we have to poll? can't we just query for council members when requested?

Reply:
Fabo said it is better to cache them :shrug:

Reply:
I said cache them using a cache directive. Let's remove the polling to reduce code complexity.

Reply:
I don't know what you mean with cache directive. Just an object, like you did for `sessionCache` ?

Reply:
No, a graphql caching directive in schema.js

Reply:
Done